# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/EditorBase.h
# Commit: 555d2ca16477
# Full Hash: 555d2ca1647762dfcdb7651a260d133782782ba7
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-21 09:52:11
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 1: Make `TextEditor` and `HTMLEditor` implement `EditorBase::InitEditorContentAndSelection` by themselves r=m_kato CLOSED TREE
#   
#   The method is enough simple, and uses bad cast from point of view of OOP.
#   Therefore, this patch make the sub classes implement the method only for each.
#   
# ==============================================================================

diff -r fb7ca98a6881 -r 555d2ca16477 editor/libeditor/EditorBase.h
--- a/editor/libeditor/EditorBase.h	Wed Sep 21 04:40:46 2022 +0300
+++ b/editor/libeditor/EditorBase.h	Wed Sep 21 00:20:26 2022 +0000
@@ -2245,13 +2245,6 @@
 
   [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult EnsureEmptyTextFirstChild();
 
-  /**
-   * InitEditorContentAndSelection() may insert a padding `<br>` element for
-   * if it's required in the anonymous `<div>` element or `<body>` element and
-   * collapse selection at the end if there is no selection ranges.
-   */
-  [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult InitEditorContentAndSelection();
-
   int32_t WrapWidth() const { return mWrapColumn; }
 
   /**